bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiBinding.java

```diff
@@ -205,8 +208,12 @@ protected void execute() {
                 }
             }
 
+        } catch (SocketTimeoutException e) {
+            logger.info("Receive Timeout on CoE socket");
         } catch (Exception e) {
-            logger.warn("Error in execute: {}, Message: {}", e.getClass().getName(), e.getMessage());
+            StringWriter w = new StringWriter();
+            e.printStackTrace(new PrintWriter(w));
```

Comment:
The correct way to log exceptions would be like:
```
logger.warn("Error in execute", e);
```
The default logback configuration will put most of the stack trace in the log this way.